### PR TITLE
Update ivy-publish snippet to use convention plugins instead of cross-configuration

### PR DESCRIPTION
**Pull request by rieske**
Updates the full example shown here to use the recommended practices: https://docs.gradle.org/nightly/userguide/publishing_ivy.html#publishing_ivy:example

### CODE REVIEW

File: subprojects/docs/src/snippets/ivy-publish/java-multi-project/kotlin/project1/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+plugins {
+    id("myproject.publishing-conventions")
+}
+
+description = "The first project"
+
+dependencies {
+    "implementation"("junit:junit:4.13")
+    "implementation"(project(":project2"))
```
Comment (jjohannes) [inline subprojects/docs/src/snippets/ivy-publish/java-multi-project/kotlin/project1/build.gradle.kts:9]: :x: These should now be `implementation` (without `"`) as the accessors are now available.

File: subprojects/docs/src/snippets/ivy-publish/java-multi-project/kotlin/project2/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+plugins {
+    id("myproject.publishing-conventions")
+}
+
+description = "The second project"
+
+dependencies {
+    "implementation"("commons-collections:commons-collections:3.2.2")
```
Comment (jjohannes) [inline subprojects/docs/src/snippets/ivy-publish/java-multi-project/kotlin/project2/build.gradle.kts:8]: :x: same as above

**Review by jjohannes — CHANGES_REQUESTED** [review]
Another good one. Thanks!

Found one thing that can be improved as well now.

**Review by jjohannes — APPROVED** [review]

**Review by donat — APPROVED** [review]
LGTM

**Pull request merged**